File: rules/vuln_samples/positive/javascript/sql_injection.js
// POSITIVE: SQL injection in Node.js
const mysql = require('mysql');

function getUser(req, res) {
    const userId = req.query.id;
    // Unsafe: template literal in query
    db.query(`SELECT * FROM users WHERE id = ${userId}`, callback);
    // Unsafe: concatenation
    pool.query("SELECT * FROM users WHERE id = " + userId, callback);
    // Unsafe: template literal with request body
    client.query(`SELECT * FROM products WHERE name = '${req.body.name}'`);
}

File: rules/vuln_samples/positive/python/sql_injection.py
# POSITIVE: SQL injection — should trigger py-sql-injection, py-sql-injection-fstring
import sqlite3
from flask import request

def get_user():
    conn = sqlite3.connect("app.db")
    cursor = conn.cursor()
    user_id = request.args.get('id')
    
    # Unsafe: string concatenation
    cursor.execute("SELECT * FROM users WHERE id = '" + user_id + "'")
    # Unsafe: f-string
    cursor.execute(f"SELECT * FROM users WHERE name = '{user_id}'")
    # Unsafe: % formatting
    cursor.execute("SELECT * FROM users WHERE id = '%s'" % user_id)
